### PR TITLE
Add user doc for local transaction

### PR DESCRIPTION
**Pull request by indikasampath2000**
## Purpose
Explain how the local transaction works in the broker. Also, add some details about the difference between AMQP and JMS client in the local transaction.

## Goals
Provide high-level details of transaction support.

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: docs/user/local-transaction.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+                                int acknowledgeMode)
+                                throws JMSException
+```
+
+But AMQP client should invoke following method to begin the transaction.
+```
+Channel channel = connection.createChannel();
+channel.txSelect();
+```
+Once transaction began, rest of the messages that are published and acknowledged kept in the broker memory until client
```
Comment (a5anka) [inline docs/user/local-transaction.md]: Maybe `After the transaction is started` is more grammatically correct in this context.

File: docs/user/local-transaction.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Local transaction
+The broker semantics of handling local transaction is based on the AMQP version 0-9-1 specification. The broker allows 
+publish and acknowledge operations to be batched into atomic units of work until client confirm it succeeds or fails by 
+doing commit or rollback. The following high-level sequence diagram depicting how the main flow of transaction works.
+
+![Local Transaction Sequence](../images/local-transaction-sequence.png)
+
+## Client library support
```
Comment (a5anka) [inline docs/user/local-transaction.md]: Shall we specify some example clients jars we can use? For example rabbitmq,  andes-client

File: docs/user/local-transaction.md (hunk 1 above)
Comment (indikasampath2000) [reply docs/user/local-transaction.md]: Corrected.

File: docs/user/local-transaction.md (hunk 2 above)
Comment (indikasampath2000) [reply docs/user/local-transaction.md]: +1. I mentioned about WSO2 andes-client and RabbitMQ amqp-client in the paragraph.

**Pull request merged**